### DISCUSSION TITLE
Testing APIs

### DISCUSSION THREAD

**Discussion by vamsibvv** [rahulvemula/SE-leasingPortal#48] [Q&A]
Hi Team, We have created user register and login apis for backend.

Please test and let us know if any issues

**Comment by vamsibvv** [comment]
API endpoints:
1. http://localhost:9010/user/ - GET
2. http://localhost:9010/user/ - POST
   Sample Payload: (dont use this . modify while testing)
    {
    "name":"vamsi",
    "email":"vbethamsetty@ufl.edu",
    "password": "vamsi"
   }
3.  http://localhost:9010/user/{email} - GET  by email
4. 2. http://localhost:9010/user/ - PUT
   Sample Payload: (dont use this . modify while testing)
    {
    "name":"vamsi",
    "email":"vbethamsetty@ufl.edu",
    "password": "vamsi"
   }

**Comment by vamsibvv** [comment]
API endpoints for leasing:

http://localhost:9010/lease/ - GET
http://localhost:9010/lease/ - POST
Sample Payload: (dont use this . modify while testing)
{
    "listingId":1,
    "userId":"1",
    "leaseStartDate": "28 Jan",
    "leaseEndDate" : "14 Feb"
}
http://localhost:9010/lease/{leaseId} - GET by leaseId
http://localhost:9010/lease/ - PUT
Sample Payload: (dont use this . modify while testing)
{
    "listingId":1,
    "userId":"1",
    "leaseStartDate": "28 Jan",
    "leaseEndDate" : "14 Feb"
}

**Comment by mitulmandaliya** [comment]
API endpoints for apartment:

http://localhost:9010/apartment/ - GET
http://localhost:9010/apartment/ - POST
Sample Payload: (dont use this . modify while testing)
{
"name":"",
"address":"",
"amenities": "one, two, three"
}
[http://localhost:9010/apartment/{apartmendId}](http://localhost:9010/apartment/%7BapartmentId%7D) - GET by apartmentId
http://localhost:9010/apartment/ - PUT
Sample Payload: (dont use this . modify while testing)
{
"name":"",
"address":"",
"amenities": ""
}

NOTE: As of now, we are storing amenities as a comma separated string

**Comment by mitulmandaliya** [comment]
API endpoints for listing:

http://localhost:9010/listing/ - GET
http://localhost:9010/listing/ - POST
Sample Payload: (dont use this . modify while testing)
{
"listingType":"bedroom",
"houseType":"1",
"rent": 500,
"userId": 1,
"isleased": true
}
[http://localhost:9010/listing/{listingId}](http://localhost:9010/listing/%7BlistingId%7D) - GET by listingId
http://localhost:9010/listing/ - PUT
Sample Payload: (dont use this . modify while testing)
{
"listingType":"bedroom",
"houseType":"1",
"rent": 500,
"userId": 1,
"isleased": true
}
